docs/en/integrations/data-ingestion/apache-spark/index.md

```diff
@@ -426,68 +436,335 @@ One of the most used data sources supported by Spark is JDBC.
+  spark.stop()
+
+}
+```
+
+Find more details on the [`s3` table function page](/docs/en/sql-reference/table-functions/s3.md).
+
+</TabItem>
+<TabItem value="Python" label="Python">
+
+```python
+from pyspark.sql import SparkSession
+
+jar_files = [
+    "jars/clickhouse-jdbc-X.X.X-SNAPSHOT-all.jar"
+]
+
+# Initialize Spark session with JARs
+spark = SparkSession.builder \
+    .appName("example") \
+    .master("local") \
+    .config("spark.jars", ",".join(jar_files)) \
+    .getOrCreate()
+
+url = "jdbc:ch://localhost:8123/default"
+user = "your_user" 
+password = "your_password"  
+query = "select * from example_table where id > 2"
+driver = "com.clickhouse.jdbc.ClickHouseDriver"
+
+df = (spark.read
+      .format('jdbc')
+      .option('driver', driver)
+      .option('url', url)
+      .option('user', user)
+      .option('password', password).option(
+    'query', query).load())
+
+df.show()
+
+```
+
+</TabItem>
+<TabItem value="SparkSQL" label="SparkSQL">
+
+```sql
+   CREATE TEMPORARY VIEW jdbcTable
+           USING org.apache.spark.sql.jdbc
+           OPTIONS (
+                   url "jdbc:ch://localhost:8123/default", 
+                   dbtable "schema.tablename",
+                   user "username",
+                   password "password",
+                   driver "com.clickhouse.jdbc.ClickHouseDriver" 
+           );
+           
+   SELECT * FROM jdbcTable;
+```
+
+</TabItem>
+</Tabs>
 
 ### Write data
 
-:::important
-As of today, you can insert data using JDBC only into existing tables. 
-:::
+
+<Tabs groupId="spark_apis">
+<TabItem value="Java" label="Java" default>
 
 ```java
-    public static void main(String[] args) {
+ public static void main(String[] args) {
         // Initialize Spark session
         SparkSession spark = SparkSession.builder().appName("example").master("local").getOrCreate();
 
         // JDBC connection details
         String jdbcUrl = "jdbc:ch://localhost:8123/default";
         Properties jdbcProperties = new Properties();
         jdbcProperties.put("user", "default");
-        jdbcProperties.put("password", "******");
+        jdbcProperties.put("password", "123456");
+
         // Create a sample DataFrame
         StructType schema = new StructType(new StructField[]{
                 DataTypes.createStructField("id", DataTypes.IntegerType, false),
                 DataTypes.createStructField("name", DataTypes.StringType, false)
         });
-        
+
         List<Row> rows = new ArrayList<Row>();
         rows.add(RowFactory.create(1, "John"));
         rows.add(RowFactory.create(2, "Doe"));
 
+
         Dataset<Row> df = spark.createDataFrame(rows, schema);
 
+        //---------------------------------------------------------------------------------------------------
+        // Write the df to ClickHouse using the jdbc method
+        //---------------------------------------------------------------------------------------------------
+
         df.write()
```

Comment:
I think we should write meaningful variables names

Reply:
Do you have any ideas for meaningful variable names? it is a standard practice to call a DataFrame df